redash/models/parameterized_query.py

```diff
@@ -29,6 +30,13 @@ def _load_result(query_id, org):
         raise QueryDetachedFromDataSourceError(query_id)
 
 
+def query_has_parameters(query_id, org):
+    try:
+        query = models.Query.get_by_id_and_org(query_id, org)
+        return bool(query.parameters)
+    except (models.NoResultFound):
```

Comment:
Wouldn't we prefer that this blows up with a 404 in such a case? Returning `False` implies that this query doesn't have parameters, but in-fact, it's a misconfigured parent query. It might make debugging harder.

Reply:
Returning `False` was just the easy option (because it will invalidate any option you provide to the parameter), I actually just wanted to avoid the error happening on the server and check what should be done in the rest of this file (currently a server error happens for misconfigured parent queries).

Rethinking about it, to better cover the misconfigured parameters at this point we can either create a new exception type or reuse `InvalidParameterError` and blow up with a 400. IMO there's probably a more robust way, but showing "Your query has misconfigured parameterers: {}" error on query execution should be helpful enough.

Reply:
I now wonder if it's worth it to add the extra logic to have a new Exception type, considering that #4707 will show no options and add a "No options available" empty state. Also I think any extra work here may end up on rework from #4312

Reply:
Updated it to consider that `NoResultFound` as an invalid parameter, the UI will still show it as required when editing. This should be enough help for identifying a problem with a query based dropdown.